### PR TITLE
Support checking additional properties for flattened templates

### LINKED ISSUE #295: Engine schema should be more restrictive and do not allow additional properties
Engine should also set `additionalProperties = false` in schemas and make the validation more restrictive. This gives users more directions and error messages when their schema structure is wrong, or has typos or bad keywords. Otherwise, the engine might just run successfully but their engine configs can be written wrong, and changes didn't get picked up by the engine.

### PR DESCRIPTION
**Pull request by umairidris**
Fix #295

### CODE REVIEW

File: internal/policygen/schema.go [hunk 1]
```diff
@@ -64,11 +64,6 @@ properties = {
 var orgPoliciesSchema = []byte(`
 title                = "Google Cloud Organization Policies Config Schema"
 additionalProperties = false
-required = [
-  "parent_type",
-  "parent_id",
-  "allowed_policy_member_customer_ids",
```
Comment (xingao267) [inline internal/policygen/schema.go:70]: do we still want `allowed_policy_member_customer_ids` to be required?

File: templates/tfengine/recipes/resources.hcl [hunk 2]
```diff
@@ -104,6 +104,13 @@ schema = {
             description = "ID of bigquery dataset."
             type        = "string"
           }
+          resource_name = {
+            description = <<EOF
+              Override for Terraform resource name. If unset, defaults to normalized dataset_id.
+              Normalization will occur based on the
```
Comment (xingao267) [inline templates/tfengine/recipes/resources.hcl:110]: based on the ...

File: internal/policygen/schema.go (hunk 1 above)
Comment (umairidris) [reply internal/policygen/schema.go:70]: Done

File: templates/tfengine/recipes/resources.hcl (hunk 2 above)
Comment (umairidris) [reply templates/tfengine/recipes/resources.hcl]: Fixed

**Pull request merged**